src/core/net/dnssd_server.cpp

```diff
@@ -164,7 +164,7 @@ void Server::ProcessQuery(Request &aRequest)
     if (mEnableUpstreamQuery && ShouldForwardToUpstream(aRequest))
     {
         Error error = ResolveByUpstream(aRequest);
-
+        mCounters.mHandledByUpstream.mForwardedQueries++;
```

Comment:
How about moving this to `ResolveByUpstream()` method?
(since this is the method that  implements the DNS upstream functionality, it should be the one that also manages and updates all the related counters).